From 81e52b65aa6767d0542d9aeee650bcca6907aa56 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Mon, 30 Aug 2021 11:43:57 +0800
Subject: [PATCH] dts: nuvoton: evb-npcm845: update device tree

1. change FIU0 CS1 flash layout
2. use dual IO for FIU0 direct Read
3. add SGMII workaround for EVB Rev.A
4. connect JTM1 to onboard CPLD

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 .../boot/dts/nuvoton/nuvoton-npcm845-evb.dts  | 74 ++++++++-----------
 .../nuvoton/nuvoton-npcm845-pincfg-evb.dtsi   | 13 ++++
 2 files changed, 45 insertions(+), 42 deletions(-)

diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
index a9b077208061..d656b212a50e 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
@@ -137,6 +137,8 @@
 		gmac0: eth@f0802000 {
 			phy-mode = "rgmii-id";
 			snps,eee-force-disable;
+			gpios = <&gpio2 14 GPIO_ACTIVE_HIGH>,
+					<&gpio2 15 GPIO_ACTIVE_HIGH>;
 			status = "okay";
 		};
 
@@ -192,58 +194,43 @@
 
 		fiu0: spi@fb000000 {
 			status = "okay";
-			spi-nor@0 {
+			spi-nor@1 {
 				compatible = "jedec,spi-nor";
 				#address-cells = <1>;
 				#size-cells = <1>;
-				reg = <0>;
+				spi-rx-bus-width = <2>;
+				spi-tx-bus-width = <2>;
+				reg = <1>;
 				spi-max-frequency = <5000000>;
-				partitions@80000000 {
+
+				partitions@88000000 {
 					compatible = "fixed-partitions";
 					#address-cells = <1>;
 					#size-cells = <1>;
-					bbuboot1@0 {
-						label = "bb-uboot-1";
-						reg = <0x0000000 0x80000>;
-						read-only;
-						};
-					bbuboot2@80000 {
-						label = "bb-uboot-2";
-						reg = <0x0080000 0x80000>;
-						read-only;
-						};
-					envparam@100000 {
-						label = "env-param";
-						reg = <0x0100000 0x40000>;
+					bmc@0{
+						label = "bmc";
+						reg = <0x000000 0x2000000>;
+					};
+					u-boot@0 {
+						label = "u-boot";
+						reg = <0x0000000 0xC0000>;
 						read-only;
-						};
-					spare@140000 {
-						label = "spare";
-						reg = <0x0140000 0xC0000>;
-						};
+					};
+					u-boot-env@100000{
+						label = "u-boot-env";
+						reg = <0x00100000 0x40000>;
+					};
 					kernel@200000 {
 						label = "kernel";
-						reg = <0x0200000 0x400000>;
-						};
-					rootfs@600000 {
-						label = "rootfs";
-						reg = <0x0600000 0x700000>;
-						};
-					spare1@D00000 {
-						label = "spare1";
-						reg = <0x0D00000 0x200000>;
-						};
-					spare2@0F00000 {
-						label = "spare2";
-						reg = <0x0F00000 0x200000>;
-						};
-					spare3@1100000 {
-						label = "spare3";
-						reg = <0x1100000 0x200000>;
-						};
-					spare4@1300000 {
-						label = "spare4";
-						reg = <0x1300000 0x0>;
+						reg = <0x0200000 0x800000>;
+					};
+					rofs@800000 {
+						label = "rofs";
+						reg = <0xa00000 0x1500000>;
+					};
+					rwfs@1f00000 {
+						label = "rwfs";
+						reg = <0x1f00000 0x100000>;
 					};
 				};
 			};
@@ -532,6 +519,9 @@
 		pinctrl-0 = <
 				&jtag2_pins
 				&lpc_pins
+				&pin70_o
+				&pin108_slew
+				&pin109_slew
 				&pin240_slew
 				&pin241_slew
 				&pin242_slew
diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-pincfg-evb.dtsi b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-pincfg-evb.dtsi
index 98622205ed79..0cdb35aa8e87 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-pincfg-evb.dtsi
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-pincfg-evb.dtsi
@@ -3,6 +3,19 @@
 
 / {
 	pinctrl: pinctrl@f0800000 {
+		pin70_o: pin70-o {
+			 pins = "GPIO70/FANIN6";
+			 bias-disable;
+			 output-high;
+		};
+		pin108_slew: pin108-slew {
+			pins = "GPIO108/SG1_MDC";
+			slew-rate = <1>;
+		};
+		pin109_slew: pin109-slew {
+			pins = "GPIO109/SG1_MDIO";
+			slew-rate = <1>;
+		};
 		pin33_slew: pin33-slew {
 			pins = "I3C4_SCL";
 			slew-rate = <1>;
-- 
2.17.1

